x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilege.java

```diff
@@ -61,6 +61,7 @@ private ManageOwnClusterPermissionCheck() {
 
         @Override
         protected boolean extendedCheck(String action, TransportRequest request, Authentication authentication) {
+            authentication.assertNotRemoteAccess();
```

Comment:
This is a cluster privilege, and the actions are all non-cross-cluster; as such it should never apply for remote access -- adding this assertion here as a sanity check. The real way to ensure nothing goes wrong here is most likely 1. enforcing a whitelist of request types that are allowed on the new remote access port, 2. excluding any non-index privileges from the role descriptors received in remote access headers.

Reply:
I don't think we need this assertion here. Otherwise by analogy, we will need it for almost all cluster privileges. Why should this be special? 

> 1. enforcing a whitelist of request types that are allowed on the new remote access port,

We will need this. The sender side already has it. So similar check is needed on the receiver side since we know that it is possible to manually issue requests to the remote access port.

> 2. excluding any non-index privileges from the role descriptors received in remote access headers.

Isn't this already done in `RBACEngine.getRemoteAccessRoleDescriptorsIntersection`?

Reply:
> Isn't this already done in RBACEngine.getRemoteAccessRoleDescriptorsIntersection?

It is, but that's on the querying cluster side -- we likely want this on the fulfilling cluster, to block manually crafted requests with unexpected privileges (e.g., cluster) from causing trouble.

Reply:
Right, we always need pair the checks on both sides because the requests can be manually crafted.